packages/eas-cli/src/build/credentials.ts

```diff
@@ -1,104 +1,32 @@
-      const hasRemote = await provider.hasRemoteAsync();
-      if (hasRemote && hasLocal) {
-        if (!(await provider.isLocalSyncedAsync())) {
-          if (nonInteractive) {
-            throw new Error(
-              `Contents of your local credentials.json for ${platform} are not the same as credentials on EAS servers. To use the desired credentials, set the "builds.${platform}.{profile}.credentialsSource" field in the credentials.json file to one of the following: "local", "remote".`
-            );
-          } else {
-            Log.log(
-              `Contents of your local credentials.json for ${platform} are not the same as credentials on EAS servers`
-            );
-          }
-
-          const { select } = await promptAsync({
-            type: 'select',
-            name: 'select',
-            message: 'Which credentials do you want to use for this build?',
-            choices: [
-              { title: 'Local credentials.json', value: CredentialsSource.LOCAL },
-              { title: 'Credentials stored on EAS servers.', value: CredentialsSource.REMOTE },
-            ],
-          });
-          return select;
-        } else {
-          return CredentialsSource.LOCAL;
-        }
-      } else if (hasLocal) {
-        logCredentials('local', provider.platform);
-        return CredentialsSource.LOCAL;
-      } else if (hasRemote) {
-        logCredentials('remote', provider.platform);
-        return CredentialsSource.REMOTE;
-      } else {
-        if (nonInteractive) {
-          throw new Error(
-            `Credentials for this app are not configured and there is no entry in credentials.json for ${platform}. Either configure credentials.json, or launch the build without "--non-interactive" flag to get a prompt to generate credentials automatically.`
-          );
-        } else if (Log.isDebug) {
-          Log.warn(
-            `Credentials for this app are not configured and there is no entry in credentials.json for ${platform}`
-          );
-        }
-
-        const confirm = await confirmAsync({
-          message: `${platform} credentials not found, generate now?`,
-        });
-        if (confirm) {
-          return CredentialsSource.REMOTE;
-        } else {
-          throw new Error(`Aborting build process, credentials are not configured for ${platform}`);
-        }
-      }
-    }
-  }
-}
-
-export async function ensureCredentialsAsync(
+export function resolveCredentialsSource(
   provider: CredentialsProvider,
-  workflow: Workflow,
-  src: CredentialsSource,
-  nonInteractive: boolean
-): Promise<CredentialsSource.LOCAL | CredentialsSource.REMOTE> {
+  src: CredentialsSource
+): CredentialsSource.LOCAL | CredentialsSource.REMOTE {
   switch (src) {
     case CredentialsSource.LOCAL:
       logCredentials('local', provider.platform);
       return CredentialsSource.LOCAL;
+    case CredentialsSource.AUTO:
+      Log.warn(`"credentialsSource": "auto" is deprecated, using "remote" instead`);
```

Comment:
throw error, maybe even on schema validation(not sure if joi have hooks like that)